services-core/src/solution_submission.rs

```diff
@@ -203,30 +197,23 @@ impl<'a> StableXSolutionSubmitting for StableXSolutionSubmitter<'a> {
         claimed_objective_value: U256,
         gas_price_cap: f64,
     ) -> Result<(), SolutionSubmissionError> {
-        // If the gas price cap is not at least the fast gas price then submitting the solution
-        // is not feasible because it would take too long.
-        match self.gas_price_estimating.estimate().await {
```

Comment:
I have removed this because we already check this when invoking the solver with min-avg-fee so it is unlikely that it fails here. Even with the check the fast gas price could still change afterwards before the transaction is confirmed so it does not guarantee anything and removing it makes the code here simpler.

Reply:
Yes the cancellation still happens.